packages/quantic/force-app/main/default/lwc/quanticRecentResultsList/__tests__/quanticRecentResultList.test.js

```diff
@@ -0,0 +1,496 @@
+          .mockImplementation(() => {});
+      });
+
+      afterEach(() => {
+        consoleErrorSpy.mockRestore();
+      });
+
+      it('should display the recent results list with the default options and no longer display the placeholder component', async () => {
+        recentResultsListState = {
+          results: exampleResults,
+          maxLength: defaultMaxLength,
+        };
+        const element = createTestComponent();
+        updateRecentResultsState();
+        await flushPromises();
+
+        const placeholder = element.shadowRoot.querySelector(
+          selectors.placeholder
+        );
+
+        expect(placeholder).toBeNull();
+
+        const recentResultsContainer = element.shadowRoot.querySelector(
+          selectors.quanticCardContainer
+        );
+        expect(recentResultsContainer).not.toBeNull();
+        expect(recentResultsContainer.title).toBe(defaultRecentResultsTitle);
+
+        const recentResultsListItems = element.shadowRoot.querySelectorAll(
+          selectors.recentResultItem
+        );
+
+        expect(recentResultsListItems.length).toBe(exampleResults.length);
+        recentResultsListItems.forEach((item, index) => {
+          const resultLink = item.querySelector(
+            selectors.quanticRecentResultLink
+          );
+          expect(resultLink.result).toEqual(mockResults[index]);
+        });
+      });
+
+      it('should use the correct initial state retrieved from the localstorage', async () => {
+        const getItemFromLocalStorageSpy = jest
+          .spyOn(utils, 'getItemFromLocalStorage')
+          .mockReturnValue(exampleResults);
+        createTestComponent();
+        await flushPromises();
+
+        expect(functionMocks.buildRecentResultsList).toHaveBeenCalledTimes(1);
+        expect(functionMocks.buildRecentResultsList).toHaveBeenCalledWith(
+          exampleEngine,
+          {
+            initialState: {results: exampleResults},
+            options: {maxLength: defaultMaxLength},
+          }
+        );
+
+        expect(getItemFromLocalStorageSpy).toHaveBeenCalledTimes(1);
+        expect(getItemFromLocalStorageSpy).toHaveBeenCalledWith(
+          localStorageKey
+        );
+      });
+
+      it('should call #setItemInLocalStorage function with the proper parameters when the recent result change in the state', async () => {
+        const setItemInLocalStorageSpy = jest.spyOn(
+          utils,
+          'setItemInLocalStorage'
+        );
+        createTestComponent();
+        recentResultsListState.results = exampleResults;
+        updateRecentResultsState();
+        await flushPromises();
+
+        expect(setItemInLocalStorageSpy).toHaveBeenCalledWith(
+          localStorageKey,
+          exampleResults
+        );
+        expect(setItemInLocalStorageSpy).toHaveBeenCalledTimes(1);
+      });
+    });
+
+    describe('with a custom #maxLength value', () => {
+      it('should set the #maxLength value in the controller', async () => {
```

Comment:
We should also test that the recent results list actually respects the maxLength property.

Add more than the maxLength results to the localStorage and checks that it doesn't render more than that.

Reply:
ahhh gotcha okay then it makes sense